modules/nf-core/gsea/gsea/main.nf

```diff
@@ -78,9 +76,71 @@ process GSEA_GSEA {
         sed -i.bak "s/\$f/${prefix}\${f}/g" *.rpt *.html && rm *.bak
     done
 
+    # Rename files so that they can be properly referenced by the output channels
+    # Function to rename files based on the given pattern
+    rename_files() {
+        local pattern=\$1
+        local exclude_patterns=\$2
+        local extension=\$3
+
+        # Find files matching the pattern but not matching the exclusion patterns
+        find . -type f -name "\$pattern" | while read -r file; do
+            # Exclude files based on the provided exclusion patterns
+            if ! echo "\$file" | grep -qE "\$exclude_patterns"; then
+                # Rename the file by adding the prefix "gene_sets_"
+                mv "\$file" "\$(dirname "\$file")/gene_sets_\$(basename "\$file")"
+            fi
+        done
+    }
+
+    # Pattern and exclusion for .tsv files
+    tsv_pattern="*.tsv"
+    tsv_exclude="gene_set_size|gsea_report|ranked_gene_list"
+
+    # Pattern and exclusion for .html files
+    html_pattern="*.html"
+    html_exclude="gsea_report|heat_map_corr_plot|index|pos_snapshot|neg_snapshot"
+
+    # Pattern and exclusion for .png files
+    png_pattern="*.png"
+    png_exclude="butterfly|enplot|global_es_histogram|gset_rnd_es_dist|heat_map|pvalues_vs_nes_plot|ranked_list_corr"
+
+    # Rename .tsv files
+    rename_files "\$tsv_pattern" "\$tsv_exclude" ".tsv"
+
+    # Rename .html files
+    rename_files "\$html_pattern" "\$html_exclude" ".html"
+
+    # Rename .png files
+    rename_files "\$png_pattern" "\$png_exclude" ".png"
```

Comment:
```suggestion
  # Prefix gene_set files
  for e in \
    tsv:gene_set_size\|gsea_report\|ranked_gene_list \
    html:gsea_report\|heat_map_corr_plot\|index\|pos_snapshot\|neg_snapshot \
    png:butterfly\|enplot\|global_es_histogram\|gset_rnd_es_dist\|heat_map\|pvalues_vs_nes_plot\|ranked_list_corr
  do
    IFS=: read -r ext exclude <<<"\$e"
    find . -type f -name "*.\$ext" | grep -Ev "\$exclude" | while IFS= read -r f
    do
      mv "$\f" "\$(dirname "\$f")/gene_sets_\$(basename "$\f")"
    done
  done
```

Could have been done a bit more concisely?

Reply:
Even though longer, I would argue that the previous version is a lot more readable. But not the hill I'm going to die on.